src/genie/libs/parser/iosxr/show_mpls.py

```diff
@@ -71,6 +71,144 @@ def cli (self, output=None):
             
         return mpls_dict
 
+# ==============================================
+# Parser for 'show mpls ldp discovery'
+# ==============================================
+class ShowMplsLdpDiscoverySchema(MetaParser):
+
+    """Schema for show mpls ldp discovery"""
+
+    schema = {
+        'vrf': {
+            Any(): {
+                Optional('local_ldp_identifier'): {
+                    Any(): {
+                        Optional('discovery_sources'): {
+                            'interfaces': {
+                                Any(): {
+                                    Optional('transport_ip_addr'): str,
+                                    Optional('xmit'): bool,
+                                    Optional('recv'): bool,
+                                    Any(): {
+                                        Any(): {
+                                            Optional('established_date'): str,
+                                            Optional('holdtime_sec'): int,
+                                            Optional('proposed_local'): int,
+                                            Optional('proposed_peer'): int,
+                                        },
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    }
+
+class ShowMplsLdpDiscovery(ShowMplsLdpDiscoverySchema):
+    
+    """Parser for show mpls ldp discovery"""
+
+    cli_command = ['show mpls ldp discovery']
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        vrf = "default"
```

Comment:
The vrf can be other value? Maybe it's better to learn the vrf from the device output?